seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/checkpoint/CheckpointCoordinator.java

```diff
@@ -285,12 +285,15 @@ private void handleCoordinatorError(CheckpointCloseReason reason, Throwable e) {
         if (checkpointCoordinatorFuture.isDone()) {
             return;
         }
-        cleanPendingCheckpoint(reason);
         updateStatus(CheckpointCoordinatorStatus.FAILED);
         checkpointCoordinatorFuture.complete(
                 new CheckpointCoordinatorState(
                         CheckpointCoordinatorStatus.FAILED, errorByPhysicalVertex.get()));
         checkpointManager.handleCheckpointError(pipelineId, false);
+        // we should wait the checkpoint manager handle the error to cancel other task by use
+        // checkpoint coordinator thread pool. So we killed the thread pool at the end of this
```

Comment:
Can you explain in detail the logical relationship of the calls?

Reply:
updated